toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1829,6 +1829,27 @@ func cleanupRpmDatabase(rootPrefix string) (err error) {
 	return
 }
 
+func RunPreInstallScripts(config configuration.SystemConfig) (err error) {
+	const squashErrors = false
+
+	for _, script := range config.PreInstallScripts {
+		ReportActionf("Running pre-install script: %s", path.Base(script.Path))
+		logger.Log.Infof("Running pre-install script: %s", script.Path)
+
+		err = shell.ExecuteLive(squashErrors, "chmod", "a+x", script.Path)
+		if err != nil {
+			return
+		}
+
+		err = shell.ExecuteLive(squashErrors, shell.ShellProgram, "-c", fmt.Sprintf("%s %s", script.Path, script.Args))
+		if err != nil {
+			return
+		}
+	}
+
+	return
+}
```

Comment:
Since this is run before the image gets created, we're running this in Mariner's build environment. Considering that, what's the use case for this functionality? Could you elaborate on that in the PR's summary as well btw.?

Reply:
Have updated the PR summary